sources/healthdog/src/main.rs

```diff
@@ -0,0 +1,138 @@
+/*!
+# Introduction
+
+Healthdog sends anonymous information about the health of a Bottlerocket host.
+It does so by sending key-value pairs as query params in an HTTP GET request.
+
+Healthdog also has the ability to check that a list of critical services is running.
+It does so using `systemctl` and reports services that are not healthy.
+
+# What it Sends
+
+### The standard set of metrics:
+
+* `sender`: the application sending the report.
+* `event`: the event that invoked the report.
+* `version`: the Bottlerocket version.
+* `variant`: the Bottlerocket variant.
+* `arch`: the machine architecture, e.g.'x86_64' or 'arm'.
+* `region`: the region the machine is running in.
+* `seed`: the seed value used to roll-out updates.
+* `version-lock`: the optional setting that locks Bottlerocket to a certain version.
+* `ignore-waves`: an update setting that allows hosts to update before their seed is reached.
+
+### Additionally, when `healthdog` sends a 'health ping', it adds:
+
+* `is-healthy`: true or false based on whether critical services are running.
+* `failed_services`: a list of critical services that have failed, if any.
```

Comment:
To keep underscores/hyphens consistent.
```suggestion
* `failed-services`: a list of critical services that have failed, if any.
```

Reply:
More confusing than I expected, trying to sort this out based on the following:

* config: e.g. healthdog-toml should use snake_case to be consistent with updog-toml.
* query parameters: snake_case because we can choose what we want and more things are snake than kebob in our world.
* settings: e.g. defaults.toml, kebob-case to be consistent with what's there now.

Reply:
Normalized to snake_case in [1083d9b](https://github.com/bottlerocket-os/bottlerocket/compare/3fae608ba56a631a4d63129c1c854916d2dd61b6..1083d9b81122e5e5bd42940472cc6201d8bd5149)